client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx

```diff
@@ -167,7 +167,7 @@ const ProcessingStep: React.FC< ProcessingStepProps > = function ( props ) {
 			// We should only trigger signup completion for signup flows, so check if we have one.
 			if ( availableFlows[ flow ] ) {
 				availableFlows[ flow ]().then( ( flowExport ) => {
-					if ( flowExport.default.isSignupFlow ) {
+					if ( flowExport.default.isSignupFlow && ! isNewSiteMigrationFlow( flow ) ) {
 						recordSignupComplete( { ...destinationState } );
```

Comment:
Might it be cleaner to have this always be triggered only when `siteId` is present and `siteCreated` is true? This feels like a more general Stepper problem, not just linked to migration flows. There seems to be a very loose expectation that the processing step will only be run for site creation, which feels like it will generate way too many signup complete events across various situations.

Reply:
> Might it be cleaner to have this always be triggered only when siteId is present and siteCreated is true?

@daledupreez  Are you suggesting ignore the `flowExport.default.isSignupFlow` flag ?  
Maybe with OR condition like:

```tsx
if( flowExport.default.isSignupFlow || siteCreated === true) {
... trigger the event
```

The issue is the flowExport.default.isSignupFlow will be always false, because  the code above:

```
	availableFlows[ flow ]().then( ( flowExport ) => {
					if ( flowExport.default.isSignupFlow ) {
```
Runs with a certain delay (promise) and when it runs, the url already has the created siteId.

Reply:
@daledupreez do you have any concern about merge this PR, we can think in a more global fixed in further one.

Reply:
I was thinking we should trigger the event any time a site is created.
```suggestion
					if ( destinationState?.siteCreated ) {
						recordSignupComplete( { ...destinationState } );
```

And I think we can land this provided we follow up and fix this logic quickly (i.e. before we forget to!).

Reply:
@daledupreez Updated on [19b1d32](https://github.com/Automattic/wp-calypso/pull/102600/commits/19b1d3206b44544acf1f009228d0abae834134f8)

Reply:
For posterity, the solution of checking `site Created` doesn't work because there are flows that do not send any information to the destination State, even if a site is created.